### PR TITLE
Allow import withouth state verification

### PR DESCRIPTION
**Pull request by arkpar**
This is another take on reverted #3942
Instead of doing header-only sync we do regular sync here and skip state enactment if parent state is not available.

### CODE REVIEW

**Comment by gavofyork** [comment]
Master merge is need to fix the test

**Comment by rphmeier** [comment]
This is higher priority than switching to async/await.

File: core/client/src/client.rs [hunk 1]
```diff
@@ -927,22 +936,39 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
...
+				}
+				if let Some(storage_changes) = storage_changes.clone() {
+					operation.op.update_storage(storage_changes.0, storage_changes.1)?;
+				}
+				if let Some(Some(changes_update)) = changes_update {
+					operation.op.update_changes_trie(changes_update)?;
+				}
+				storage_changes
+			},
+			_ => Default::default()
```
Comment (rphmeier) [inline core/client/src/client.rs]:
```suggestion
			_ => None,
```

I'd prefer explicitness for readability

File: core/finality-grandpa/src/tests.rs [hunk 2]
```diff
@@ -984,6 +985,7 @@ fn allows_reimporting_change_blocks() {
 			bad_justification: false,
 			needs_finality_proof: false,
 			is_new_best: true,
+			header_only: false,
```
Comment (rphmeier) [inline core/finality-grandpa/src/tests.rs]: why is any of this exposed to the consensus engine (`header_only`, `allow_missing_state`), which only cares about headers?

Comment (arkpar) [reply core/finality-grandpa/src/tests.rs]: Same reasons as #3623. Fixing the pipeline is out of scope of this PR.

**Pull request merged**